shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/GpuBroadcastHashJoinExec.scala

```diff
@@ -96,12 +93,15 @@ case class GpuBroadcastHashJoinExec(
     condition: Option[Expression],
     left: SparkPlan,
     right: SparkPlan) extends BinaryExecNode with GpuHashJoin {
+  import GpuMetric._
 
-  override lazy val additionalMetrics: Map[String, SQLMetric] = Map(
-    "joinOutputRows" -> SQLMetrics.createMetric(sparkContext, "join output rows"),
-    "streamTime" -> SQLMetrics.createNanoTimingMetric(sparkContext, "stream time"),
-    "joinTime" -> SQLMetrics.createNanoTimingMetric(sparkContext, "join time"),
-    "filterTime" -> SQLMetrics.createNanoTimingMetric(sparkContext, "filter time"))
+  override lazy val outputRowsLevel: MetricsLevel = ESSENTIAL_LEVEL
+  override lazy val outputBatchesLevel: MetricsLevel = MODERATE_LEVEL
```

Comment:
Curious, why are these `lazy`?  Comment applies to many other places where this was copied.

Reply:
I made it lazy so on the chance that we might want to calculate it later. If you want me to switch it over to just be a val I can.

Reply:
Yeah these should all not be lazy.  If some exec in the future wants to calculate this dynamically then it can do a lazy override, but no need to apply this to all cases.  I'm also not sure why we would want a metric to sometimes be essential vs. moderate etc.

Reply:
You cannot do a lazy override to a non-lazy value.  They have to match, but we can cross that bridge when we come to it.